roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+import h5py
+from lazy_ops import DatasetView
+from ...segmentationextractor import SegmentationExtractor
+from ...extraction_tools import _pixel_mask_extractor
+
+class CaimanSegmentationExtractor(SegmentationExtractor):
+    """
+    This class inherits from the SegmentationExtractor class, having all
+    its funtionality specifically applied to the dataset output from
+    the \'CNMF-E\' ROI segmentation method.
+    """
+    extractor_name = 'CaimanSegmentation'
+    installed = True  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, filepath):
+        """
+        Parameters
+        ----------
+        filepath: str
+            The location of the folder containing caiman *.hdmf output file.
+        """
+        SegmentationExtractor.__init__(self)
+        self.filepath = filepath
+        self._dataset_file = self._file_extractor_read()
+        self._roi_response = self._trace_extractor_read('F_dff')
+        self._masks, self._mask_roi_ids, self._mask_ids = self._image_mask_sparse_read()
+        self._roi_response_dict = {'Fluorescence': self._roi_response,
+                                   'Neuropil':self._trace_extractor_read('C'),
+                                   'Deconvolved':self._trace_extractor_read('S')}
+
+        self._raw_movie_file_location = self._dataset_file['params']['data']['fnames'][0].decode('utf-8')
+        self._sampling_frequency = self._dataset_file['params']['data']['fr'].value
+        # file close:
+        # self._file_close()
+
+    def _file_close(self):
+        self._dataset_file.close()
+
+    def _file_extractor_read(self):
+        f = h5py.File(self.filepath, 'r')
+        return f
+
+    def _image_mask_sparse_read(self):
+        roi_ids = self._dataset_file['estimates']['A']['indices']
+        masks = self._dataset_file['estimates']['A']['data']
+        ids = self._dataset_file['estimates']['A']['indptr']
+        return masks, roi_ids, ids
+
+    def _trace_extractor_read(self, field):
+        extracted_signals = self._dataset_file['estimates'][field] # lazy read dataset)
```

Comment:
it would be better to use lazy_ops to make this lazy

Reply:
`extracted_signals `is a hdf5 dataset class. I assume that is not loaded into memory unless sliced etc, is that correct? 
If true, does calling `lazy_ops.DatasetView `over that (`DatasetView(extracted_signals)`) have further benefits?

Reply:
oh, `field` is a string. OK, I misunderstood, this looks good.